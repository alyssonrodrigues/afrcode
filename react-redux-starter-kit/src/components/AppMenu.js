import React, { Component } from 'react'
import { withStyles } from '@material-ui/core/styles'
import { connect } from 'react-redux'
import IconButton from '@material-ui/core/IconButton'
import Drawer from '@material-ui/core/Drawer'
import Divider from '@material-ui/core/Divider'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import ChevronRightIcon from '@material-ui/icons/ChevronRight'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'

import styles from './AppMenuStyles'
import { getMenuItems } from '../util/applicationContext'

export class AppMenu extends Component {
  renderMainMenuItems () {
    const { classes, authentication, handleMainMenuClose } = this.props
    return authentication && authentication.isUserAuthenticated && (
      <div className={classes.menuList}>
        <List component='nav'>
          {getMenuItems().map(it => (
            <ListItem button key={it.label} onClick={() => handleMainMenuClose(it.path)}>
              <ListItemText primary={it.label} />
            </ListItem>
          ))}
          <Divider />
          <ListItem button key='Logout' onClick={() => handleMainMenuClose('/logout')}>
            <ListItemText primary='Sair' />
          </ListItem>
        </List>
      </div>
    )
  }

  render () {
    const {
      classes,
      theme,
      mainMenu: { mainMenuOpen },
      handleMainMenuClose
    } = this.props
    return (
      <Drawer
        variant='persistent'
        anchor='left'
        open={mainMenuOpen}
        PaperProps={{style: {minHeight: '100%'}}}
        classes={{ paper: classes.drawerPaper }}>
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleMainMenuClose}>
            {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
          </IconButton>
        </div>
        <Divider />
        {this.renderMainMenuItems()}
      </Drawer>
    )
  }
}

const materialUIEnhanced = withStyles(styles, { withTheme: true })(AppMenu)

const mapStateToProps = ({ authentication, mainMenu }) => ({ authentication, mainMenu })

export default connect(mapStateToProps)(materialUIEnhanced)
