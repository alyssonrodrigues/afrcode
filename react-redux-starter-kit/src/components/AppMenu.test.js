import React from 'react'
import { shallow } from 'enzyme'

import { AppMenu } from './AppMenu'

describe('AppMenu', () => {
  it('AppMenu render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <AppMenu classes={{}} theme={{}}
        authentication={{isUserAuthenticated: true}}
        mainMenuOpen={jest.fn()} handleMainMenuClose={jest.fn()} />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
