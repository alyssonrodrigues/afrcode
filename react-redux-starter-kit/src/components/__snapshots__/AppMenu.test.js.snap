// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu AppMenu render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppMenu
    authentication={
        Object {
            "isUserAuthenticated": true,
          }
    }
    classes={Object {}}
    handleMainMenuClose={[Function]}
    mainMenuOpen={[Function]}
    theme={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "PaperProps": Object {
        "style": Object {
          "minHeight": "100%",
        },
      },
      "anchor": "left",
      "children": Array [
        <div
          className={undefined}
>
          <WithStyles(IconButton)
                    onClick={[Function]}
          >
                    <pure(ChevronLeftIcon) />
          </WithStyles(IconButton)>
</div>,
        <WithStyles(Divider) />,
        <div
          className={undefined}
>
          <WithStyles(List)
                    component="nav"
          >
                    <WithStyles(ListItem)
                              button={true}
                              onClick={[Function]}
                    >
                              <WithStyles(ListItemText)
                                        primary="Menu item 1"
                              />
                    </WithStyles(ListItem)>
                    <WithStyles(ListItem)
                              button={true}
                              onClick={[Function]}
                    >
                              <WithStyles(ListItemText)
                                        primary="Menu item 2"
                              />
                    </WithStyles(ListItem)>
                    <WithStyles(ListItem)
                              button={true}
                              onClick={[Function]}
                    >
                              <WithStyles(ListItemText)
                                        primary="Menu item N"
                              />
                    </WithStyles(ListItem)>
                    <WithStyles(Divider) />
                    <WithStyles(ListItem)
                              button={true}
                              onClick={[Function]}
                    >
                              <WithStyles(ListItemText)
                                        primary="Sair"
                              />
                    </WithStyles(ListItem)>
          </WithStyles(List)>
</div>,
      ],
      "classes": Object {
        "paper": undefined,
      },
      "open": [Function],
      "variant": "persistent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(IconButton)
            onClick={[Function]}
>
            <pure(ChevronLeftIcon) />
</WithStyles(IconButton)>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(ChevronLeftIcon) />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(List)
            component="nav"
>
            <WithStyles(ListItem)
                        button={true}
                        onClick={[Function]}
            >
                        <WithStyles(ListItemText)
                                    primary="Menu item 1"
                        />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
                        button={true}
                        onClick={[Function]}
            >
                        <WithStyles(ListItemText)
                                    primary="Menu item 2"
                        />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
                        button={true}
                        onClick={[Function]}
            >
                        <WithStyles(ListItemText)
                                    primary="Menu item N"
                        />
            </WithStyles(ListItem)>
            <WithStyles(Divider) />
            <WithStyles(ListItem)
                        button={true}
                        onClick={[Function]}
            >
                        <WithStyles(ListItemText)
                                    primary="Sair"
                        />
            </WithStyles(ListItem)>
</WithStyles(List)>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              Array [
                <WithStyles(ListItem)
                  button={true}
                  onClick={[Function]}
>
                  <WithStyles(ListItemText)
                                    primary="Menu item 1"
                  />
</WithStyles(ListItem)>,
                <WithStyles(ListItem)
                  button={true}
                  onClick={[Function]}
>
                  <WithStyles(ListItemText)
                                    primary="Menu item 2"
                  />
</WithStyles(ListItem)>,
                <WithStyles(ListItem)
                  button={true}
                  onClick={[Function]}
>
                  <WithStyles(ListItemText)
                                    primary="Menu item N"
                  />
</WithStyles(ListItem)>,
              ],
              <WithStyles(Divider) />,
              <WithStyles(ListItem)
                button={true}
                onClick={[Function]}
>
                <WithStyles(ListItemText)
                                primary="Sair"
                />
</WithStyles(ListItem)>,
            ],
            "component": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Menu item 1",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <WithStyles(ListItemText)
                  primary="Menu item 1"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Menu item 1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Menu item 2",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <WithStyles(ListItemText)
                  primary="Menu item 2"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Menu item 2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Menu item N",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <WithStyles(ListItemText)
                  primary="Menu item N"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Menu item N",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Logout",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <WithStyles(ListItemText)
                  primary="Sair"
/>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "primary": "Sair",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "PaperProps": Object {
          "style": Object {
            "minHeight": "100%",
          },
        },
        "anchor": "left",
        "children": Array [
          <div
            className={undefined}
>
            <WithStyles(IconButton)
                        onClick={[Function]}
            >
                        <pure(ChevronLeftIcon) />
            </WithStyles(IconButton)>
</div>,
          <WithStyles(Divider) />,
          <div
            className={undefined}
>
            <WithStyles(List)
                        component="nav"
            >
                        <WithStyles(ListItem)
                                    button={true}
                                    onClick={[Function]}
                        >
                                    <WithStyles(ListItemText)
                                                primary="Menu item 1"
                                    />
                        </WithStyles(ListItem)>
                        <WithStyles(ListItem)
                                    button={true}
                                    onClick={[Function]}
                        >
                                    <WithStyles(ListItemText)
                                                primary="Menu item 2"
                                    />
                        </WithStyles(ListItem)>
                        <WithStyles(ListItem)
                                    button={true}
                                    onClick={[Function]}
                        >
                                    <WithStyles(ListItemText)
                                                primary="Menu item N"
                                    />
                        </WithStyles(ListItem)>
                        <WithStyles(Divider) />
                        <WithStyles(ListItem)
                                    button={true}
                                    onClick={[Function]}
                        >
                                    <WithStyles(ListItemText)
                                                primary="Sair"
                                    />
                        </WithStyles(ListItem)>
            </WithStyles(List)>
</div>,
        ],
        "classes": Object {
          "paper": undefined,
        },
        "open": [Function],
        "variant": "persistent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(IconButton)
              onClick={[Function]}
>
              <pure(ChevronLeftIcon) />
</WithStyles(IconButton)>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(ChevronLeftIcon) />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)
              component="nav"
>
              <WithStyles(ListItem)
                            button={true}
                            onClick={[Function]}
              >
                            <WithStyles(ListItemText)
                                          primary="Menu item 1"
                            />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                            button={true}
                            onClick={[Function]}
              >
                            <WithStyles(ListItemText)
                                          primary="Menu item 2"
                            />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                            button={true}
                            onClick={[Function]}
              >
                            <WithStyles(ListItemText)
                                          primary="Menu item N"
                            />
              </WithStyles(ListItem)>
              <WithStyles(Divider) />
              <WithStyles(ListItem)
                            button={true}
                            onClick={[Function]}
              >
                            <WithStyles(ListItemText)
                                          primary="Sair"
                            />
              </WithStyles(ListItem)>
</WithStyles(List)>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                Array [
                  <WithStyles(ListItem)
                    button={true}
                    onClick={[Function]}
>
                    <WithStyles(ListItemText)
                                        primary="Menu item 1"
                    />
</WithStyles(ListItem)>,
                  <WithStyles(ListItem)
                    button={true}
                    onClick={[Function]}
>
                    <WithStyles(ListItemText)
                                        primary="Menu item 2"
                    />
</WithStyles(ListItem)>,
                  <WithStyles(ListItem)
                    button={true}
                    onClick={[Function]}
>
                    <WithStyles(ListItemText)
                                        primary="Menu item N"
                    />
</WithStyles(ListItem)>,
                ],
                <WithStyles(Divider) />,
                <WithStyles(ListItem)
                  button={true}
                  onClick={[Function]}
>
                  <WithStyles(ListItemText)
                                    primary="Sair"
                  />
</WithStyles(ListItem)>,
              ],
              "component": "nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Menu item 1",
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": <WithStyles(ListItemText)
                    primary="Menu item 1"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Menu item 1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Menu item 2",
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": <WithStyles(ListItemText)
                    primary="Menu item 2"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Menu item 2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Menu item N",
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": <WithStyles(ListItemText)
                    primary="Menu item N"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Menu item N",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Logout",
                "nodeType": "class",
                "props": Object {
                  "button": true,
                  "children": <WithStyles(ListItemText)
                    primary="Sair"
/>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "primary": "Sair",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
